### PR TITLE
feat(data_structures): modify tally transaction field rewarded_witnes…

### LINKED ISSUE #1091: Change "rewarded_witnesses" field in TallyTransaction by "slashed_witnesses"
This derives from the upcoming WIP on "Collateralization of Witnessing Activity".

### PR DESCRIPTION
**Pull request by lrubiorod**
…ses by slashed_witnesses

Close #1091

### CODE REVIEW

File: node/src/actors/chain_manager/mod.rs [hunk 1]
```diff
@@ -738,10 +738,14 @@ impl ReputationInfo {
         self.alpha_diff += Alpha(u32::from(replication_factor));
 
         // Set of pkhs which were rewarded in the tally transaction
```
Comment (tmpolaczyk) [inline node/src/actors/chain_manager/mod.rs]:
```suggestion
        // Set of pkhs which were slashed in the tally transaction
```

File: node/src/actors/chain_manager/mod.rs [hunk 2]
```diff
@@ -738,10 +738,14 @@ impl ReputationInfo {
...
-        let honest_witnesses: HashSet<_> = tally_transaction.rewarded_witnesses.iter().collect();
+        let slashed_witnesses: HashSet<_> = tally_transaction.slashed_witnesses.iter().collect();
         for pkh in commits.keys() {
             // If the identity behaved honestly, it must have received a reward.
```
Comment (tmpolaczyk) [inline node/src/actors/chain_manager/mod.rs]:
```suggestion
            // If the identity was slashed, it must not receive a reward.
```

File: validations/src/validations.rs [hunk 3]
```diff
@@ -392,8 +392,14 @@ pub fn validate_consensus(
     miner_tally: &[u8],
     tally: &RADTally,
     non_error_min: f64,
-    num_commits: usize,
+    commit_pkhs: Vec<&PublicKeyHash>,
 ) -> Result<HashSet<PublicKeyHash>, failure::Error> {
+    let num_commits = commit_pkhs.len();
+    let mut dishonest_hs = HashSet::default();
```
Comment (tmpolaczyk) [inline validations/src/validations.rs]:
```suggestion
    let mut dishonest_hs: HashSet<_> = commit_pkhs.iter().cloned().collect();
```

File: validations/src/validations.rs [hunk 4]
```diff
@@ -392,8 +392,11 @@ pub fn validate_consensus(
     miner_tally: &[u8],
     tally: &RADTally,
     non_error_min: f64,
-    num_commits: usize,
+    commit_pkhs: Vec<&PublicKeyHash>,
 ) -> Result<HashSet<PublicKeyHash>, failure::Error> {
+    let num_commits = commit_pkhs.len();
+    let mut dishonest_hs: HashSet<PublicKeyHash> = commit_pkhs.iter().cloned().cloned().collect();
```
Comment (tmpolaczyk) [inline validations/src/validations.rs]: Double cloned... :laughing:

Comment (lrubiorod) [reply validations/src/validations.rs]: magic

**Pull request merged**